packages/trader/src/Modules/Trading/Containers/trade-params-mobile.tsx

```diff
@@ -224,7 +224,28 @@ const TradeParamsMobile = observer(
         h_duration,
         d_duration,
     }: TTradeParamsMobile) => {
-        const { basis_list, basis, expiry_epoch, is_turbos, is_vanilla } = useTraderStore();
+        const {
+            basis_list,
+            basis,
+            expiry_epoch,
+            is_turbos,
+            is_vanilla,
+            duration_unit: selected_duration_unit,
+        } = useTraderStore();
+        const toggled_duration_tab_idx = duration_units_list.findIndex(d => d.value === duration_unit);
+        const selected_duration_tab_idx = duration_units_list.findIndex(d => d.value === selected_duration_unit);
+
+        React.useEffect(() => {
+            if (has_duration_error) {
+                setSelectedDuration(duration_unit, makeGetDefaultDuration(0, 'default')(duration_unit));
+            } else {
+                setSelectedDuration(duration_unit, duration_value);
+            }
+            if (toggled_duration_tab_idx === -1)
+                setSelectedDuration(selected_duration_unit, makeGetDefaultDuration(0, 'default')(duration_unit));
+            setDurationTabIdx(toggled_duration_tab_idx === -1 ? selected_duration_tab_idx : toggled_duration_tab_idx);
```

Comment:
```
const defaultDuration = makeGetDefaultDuration(0, 'default')(duration_unit);
setSelectedDuration(duration_unit, has_duration_error ? duration_value : defaultDuration);
if (toggled_duration_tab_idx === -1) setSelectedDuration(selected_duration_unit, defaultDuration)
```
Maybe it'll be better?
Also, do we need `has_duration_error`, `toggled_duration_tab_id`x in array with deps?

Reply:
Please, ignore if not relevant

Reply:
i didnt add has_duration_error because i want to show Error with red text whenever there is error. If i add has_duration_error in dependency this effect would not show. Instead we'd just replace the input with default value.